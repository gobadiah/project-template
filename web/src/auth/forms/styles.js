import { bool } from 'prop-types';
import React from 'react';
import styled, { css } from 'react-emotion';

import { ClearButton, Flex, Line } from '~/styles';
import { RoundButton } from '~/components';

import facebook from './images/facebook.png';
import google from './images/google.png';

export const AccountFirstRow = styled(Flex)`
  margin-top: 23px;
  align-items: center;
`;

export const AccountRow = styled(Flex)`
  margin-top: 23px;
  align-items: center;
  & > * {
    flex-grow: 1;
  }
  & > div:nth-child(2) {
    margin-left: 16px;
  }
`;

export const AccountLastRow = styled(AccountRow)`
  margin-bottom: 42px;
`;

export const AccountTightRow = styled(AccountRow)`
  margin-top: 15px;
  & > * {
    flex-grow: 1;
  }
`;

export const AccountPictureLabel = styled('label')`
  margin-left: 25px;
`;

export const AccountRequiredFields = styled('div')`
  align-self: right;
  text-align: right;
  width: 100%;
  height: 27px;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.69;
  letter-spacing: normal;
  color: var(--dark-blue-grey);
  margin: 0;
  padding: 0;
`;

export const AccountLine = styled(Line)`
  margin-top: 19px;
  margin-bottom: 5px;
`;

export const AccountSocialDiv = styled(Flex)`
  margin-left: auto;
`;

export const AccountSubmit = styled(RoundButton)`
  width: 288px;
`;

export const AccountReset = styled(ClearButton)`
  width: 200px;
  height: 27px;
  font-size: 20px;
  font-weight: bold;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.35;
  letter-spacing: normal;
  text-align: center;
  text-decoration: underline;
  color: var(--dark-blue-grey);
`;

export const uploaderClassName = css`
  width: 0.1px;
  height: 0.1px;
  opacity: 0;
  overflow: hidden;
  position: absolute;
  z-index: -1;
`;


const size = 20;

const linkedClassName = css`
  opacity: 1;
`;

const common = css`
  width: 42px;
  height: 42px;
  border-radius: 21px;
  opacity: 0.5;
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const GoogleButton = ({ linked }) => (
  <ClearButton
    css={`
      background-color: var(--pale-red);
      ${common}
      ${linked && linkedClassName}
    `}
  >
    <img
      src={google}
      alt=''
      css={`
        max-width: ${size}px;
        max-height: ${size}px;
      `}
    />
  </ClearButton>
);

GoogleButton.propTypes = {
  linked: bool.isRequired,
};

export const FacebookButton = ({ linked }) => (
  <ClearButton
    css={`
      background-color: var(--dull-blue);
      ${common}
      ${linked && linkedClassName}
    `}
  >
    <img
      src={facebook}
      alt=''
      css={`
        max-width: ${size}px;
        max-height: ${size}px;
      `}
    />
  </ClearButton>
);

FacebookButton.propTypes = {
  linked: bool.isRequired,
};

export const signInClassName = css`
  width: 520px;
  & > * {
    margin-bottom: 23px;
  }
`;

export const ForgottenPassword = styled('a')`
  font-size: 20px;
  font-weight: bold;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.35;
  letter-spacing: normal;
  text-align: left;
  cursor: pointer;
  text-decoration: underline;
  color: var(--dark-blue-grey);
`;

export const BottomContainer = styled(Flex)`
  justify-content: space-around;
  align-items: center;
`;
