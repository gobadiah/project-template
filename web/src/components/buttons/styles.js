import styled, { css } from 'react-emotion';

export const Button = styled('button')`
  width: 249px;
  height: 74px;
  border-radius: 36.5px;
  display: flex;
  align-items: center;
  justify-content: space-around;
  position: relative;
  font-size: 26px;
  font-weight: bold;
  line-height: 1.04;
  text-align: center;
  color: var(--white);
  cursor: pointer;
  outline: none;
  border: none;
  background-color: var(--dark-blue-grey);
`;

export const arrowRight = css`
  width: 20px;
  height: 11px;
  position: absolute;
  right: 17px;
`;

export const imgLeft = css`
  max-width: 20px;
  max-height: 20px;
  position: absolute;
  left: 24px;
`;

export const google = css`
  background-color: var(--pale-red);
`;

export const facebook = css`
  background-color: var(--dull-blue);
`;
