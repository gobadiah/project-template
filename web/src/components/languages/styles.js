import styled, { css } from 'react-emotion';

export const size = `
  width: 220px;
  height: 100px;
`;

export const Menu = styled('div')`
  position: relative;
  ${size};
`;

export const SubMenu = styled('div')`
 visibility: hidden;
 opacity: 0;
 top: 100%;
 left: 0;
 width: 100%;
 transform: translateY(-2em);
 z-index: -1;
 transition: all 0.3s ease-in-out 0s, visibility 0s linear 0.3s, z-index 0s linear 0.01s;

 ${Menu}:hover &,
 ${Menu}:focus &,
 ${Menu}:focus-within & {
  visibility: visible; /* shows languages-submenu */
  opacity: 1;
  z-index: 1;
  transform: translateY(0%);
  transition-delay: 0s, 0s, 0.3s;
}
`;

export const button = css`
  background-color: Transparent;
  border: none;
  cursor: pointer;
  outline: none;
  overflow: hidden;
  padding: 0;
  ${size}
`;
