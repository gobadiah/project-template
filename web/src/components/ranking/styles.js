import styled, { css } from 'react-emotion';

import bgImage from './images/rankFullBG.png';

const ratio = 79 / 252;
const width = 251;

const common = css`
  color: var(--white);
  position: absolute;
  text-align: center;
`;

export const Container = styled('div')`
  position: relative;
  background-image: url(${bgImage});
  width: ${width}px;
  height: ${ratio * width}px;
`;

export const Main = styled('div')`
  ${common}
  width: 118px;
  font-weight: bold;
  font-size: 50px;
  line-height: 59px;
  top: 0px;
  left: 25px;
`;

export const Secondary = styled('div')`
  ${common}
  width: 82px;
  font-size: 24px;
  line-height: 29px;
  bottom: 15px;
  right: 18px;
`;
