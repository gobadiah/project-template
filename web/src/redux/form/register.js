import React from 'react';

import { reduxForm } from 'redux-form';

import {
  BirthdayField,
  EmailField,
  FirstNameField,
  LastNameField,
  PasswordField,
} from './fields';
import { BaseForm } from './utils';

class RegisterForm extends BaseForm {
  getFields = () => [
    <EmailField key='email' />,
    <PasswordField
      key='password'
      autoComplete='new-password'
    />,
    <PasswordField
      key='confirm_password'
      name='confirm_password'
      label='Password confirmation'
      autoComplete='new-password'
    />,
    <FirstNameField key='first_name' />,
    <LastNameField key='last_name' />,
    <BirthdayField key='birthday' />,
  ];
}

export { RegisterForm as PlainRegisterForm };

export default reduxForm({
  form: 'RegisterForm',
})(RegisterForm);
