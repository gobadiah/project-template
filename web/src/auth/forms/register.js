import { css } from 'react-emotion';
import { reduxForm } from 'redux-form';
import React from 'react';

import {
  BirthdayField,
  EmailField,
  FemaleField,
  FirstNameField,
  LastNameField,
  LeftHandedField,
  MaleField,
  PasswordField,
  RankingField,
  RightHandedField,
  TennisClubField,
} from '~/components/forms/fields';
import { Flex } from '~/styles';
import { Form } from '~/components/forms';
import { PureComponent } from '~/components/base';
import { RoundButton } from '~/components';

class RegisterForm extends PureComponent {
  render() {
    const { t } = this.context;

    return (
      <Form {...this.props}>
        <Flex>
          <FirstNameField />
          <LastNameField />
        </Flex>
        <EmailField />
        <Flex>
          <PasswordField autoComplete='new-password' />
          <PasswordField
            name='confirm_password'
            label='Password confirmation'
            autoComplete='new-password'
          />
        </Flex>
        <Flex>
          <BirthdayField />
          <RankingField />
        </Flex>
        <TennisClubField />
        <Flex css='justify-content: space-between'>
          <MaleField inputClassName={css`width: 282px`} />
          <FemaleField inputClassName={css`width: 282px`} />
        </Flex>
        <Flex css='justify-content: space-between'>
          <LeftHandedField inputClassName={css`width: 282px`} />
          <RightHandedField inputClassName={css`width: 282px`} />
        </Flex>
        <RoundButton text={t('I create my account')} className={css`width: 100%`} />
      </Form>
    );
  }
}

export { RegisterForm as PlainRegisterForm };

export default reduxForm({
  form: 'RegisterForm',
})(RegisterForm);
