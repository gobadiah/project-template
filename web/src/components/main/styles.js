import styled from 'react-emotion';

import { FlexColumn } from '~/styles';

// eslint-disable-next-line import/prefer-default-export
export const Container = styled(FlexColumn)`
  align-items: center;
  margin: auto;
  margin-top: 49px;
  text-align: center;
  border-radius: 5px;
  padding-left: 38px;
  padding-right: 38px;
  background-color: rgba(255, 255, 255, 0.9);
  box-sizing: border-box;
  ${props => props.width && `width: ${props.width}px;`}
`;
