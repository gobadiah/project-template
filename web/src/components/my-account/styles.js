import styled from 'react-emotion';

import { Flex } from '~/styles';

export const Container = styled(Flex)`
  align-items: center;
`;

export const A = styled('a')`
  width: 147px;
  height: 29px;
  margin-left: 11px;
  font-size: 24px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.21;
  letter-spacing: normal;
  cursor: pointer;
  text-align: left;
  color: var(--white);
`;
