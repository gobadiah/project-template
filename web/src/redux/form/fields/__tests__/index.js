import 'jsdom-global/register';

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { css } from 'react-emotion';

import { Flex, FlexColumn } from '~/styles';

import Field from '..';

const props = {
  autoComplete: 'email',
  className: css`
  color: red;
`,
  cols: 64,
  formatError: jest.fn(e => e),
  formatWarning: jest.fn(w => w),
  input: {},
  inputClassName: css`
  color: blue;
`,
  label: 'this is a label',
  placeholder: 'support@example.com',
  meta: {
    error: 'Nah this I can\'t do',
    touched: true,
    warning: 'Be careful with that',
  },
  rows: 4,
  t: jest.fn(x => x),
  type: 'email',
};

describe('Field', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<Field {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('shallow render should contain elements', () => {
    const wrapper = shallow(<Field {...props} />);
    expect(wrapper.contains(<div>Hello world</div>)).toBeFalsy();
  });

  it('should use a Flex if column is false', () => {
    const wrapper = shallow(<Field {...props} />);
    expect(wrapper.find(FlexColumn)).toHaveLength(0);
    expect(wrapper.find(Flex)).toHaveLength(1);
  });

  it('should use a FlexColumn if column is true', () => {
    const wrapper = shallow(<Field {...props} column />);
    expect(wrapper.find(FlexColumn)).toHaveLength(1);
    expect(wrapper.find(Flex)).toHaveLength(0);
  });

  it('should use a textarea if type is set to it', () => {
    const wrapper = shallow(<Field {...props} type='textarea' />);
    expect(wrapper.find('textarea')).toHaveLength(1);
    expect(wrapper.find('input')).toHaveLength(0);
  });
});
