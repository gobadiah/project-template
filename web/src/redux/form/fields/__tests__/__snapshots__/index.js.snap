// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field should match snapshot 1`] = `
.emotion-0 {
  color: blue;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: red;
}

<div
  className="emotion-1 emotion-2"
>
  <label
    htmlFor={undefined}
  >
    this is a label
  </label>
  <input
    autoComplete="email"
    className="emotion-0"
    placeholder="support@example.com"
    type="email"
  />
  <div>
    Nah this I can't do
  </div>
  <div>
    Be careful with that
  </div>
</div>
`;
